client/src/components/Tabs/Tabs.js

```diff
@@ -1,139 +1,31 @@
-   *
-   * @returns {string}
-   */
-  getDefaultActiveKey() {
-    let active = null;
-
-    if (typeof this.props.defaultActiveKey === 'string') {
-      const activeChild = React.Children.toArray(this.props.children)
-        .find((child) => child.props.name === this.props.defaultActiveKey);
-
-      if (activeChild) {
-        active = activeChild.props.name;
-      }
-    }
-
-    if (typeof active !== 'string') {
-      React.Children.forEach(this.props.children, (child) => {
-        if (typeof active !== 'string') {
-          active = child.props.name;
-        }
-      });
-    }
-
-    return active;
-  }
-
-  toggle(activeTab) {
-    if (this.props.activeTab !== activeTab) {
-      this.props.activateTab(activeTab);
-    }
-  }
-
-  /**
-   * Render an individual link for the tabset
-   *
-   * @param {object} child
-   * @returns {Component}
-   */
-  renderTab(child) {
-    if (child.props.title === null) {
-      return null;
-    }
-
-    const currentTab = this.props.activeTab || this.getDefaultActiveKey();
-
-    const classNames = classnames({
-      active: currentTab === child.props.name,
-      [child.props.tabClassName]: child.props.tabClassName,
-    });
-
-    return (
-      <NavItem>
-        <NavLink
-          onClick={() => (this.toggle(child.props.name))}
-          disabled={child.props.disabled}
-          className={classNames}
-        >
-          {child.props.title}
-        </NavLink>
-      </NavItem>
-    );
-  }
-
-  /**
-   * Builds the tabset navigation links, will hide the links if there is only one child
-   *
-   * @returns {Component}
-   */
-  renderNav() {
-    const tabs = React.Children
-      .map(this.props.children, this.renderTab);
-
-    if (tabs.length <= 1) {
-      return null;
-    }
-
-    return (
-      <Nav tabs role="tablist">
-        {tabs}
-      </Nav>
-    );
-  }
-
-  render() {
-    const { hideNav, children, activeTab } = this.props;
-
-    const containerProps = this.getContainerProps();
-    const nav = hideNav ? null : this.renderNav();
-
-    return (
-      <div {...containerProps}>
-        <div className="wrapper">
-          {nav}
-          <TabContent activeTab={activeTab || this.getDefaultActiveKey()}>
+import TabNav from './TabNav';
+import getDefaultActiveKey from './getDefaultActiveKey';
+import { TabContext } from 'hooks/useTabContext';
+
+function Tabs({
+  hideNav, children, activeTab, className, extraClass, id, activateTab, defaultActiveKey
+}) {
+  const containerProps = { className: classnames([className, extraClass]), id };
+  const currentTab = activeTab || getDefaultActiveKey(defaultActiveKey, children);
+
+  return (
+    <div {...containerProps}>
+      <div className="wrapper">
+        {!hideNav && <TabNav currentTab={currentTab} onToggle={activateTab}>{children}</TabNav>}
+        <TabContent activeTab={currentTab}>
+          <TabContext.Provider value={{ activeTab: currentTab }}>
```

Comment:
This context bit is the only real change here.

Reply:
Should `<TabContext.Provider>` wrap both `<TabNav>` + `<TabContent>`?  I'm very far from an expert, I'm just looking at https://reactjs.org/docs/context.html#updating-context-from-a-nested-component and it seems like `<TabContext.Provider>` should wrap everything and there needs to be a `<TabContext.Consumer>` that at least in the example in the react docs, seems like it broadcasts up.  So the `<TabNav>` (or `<Tab>`?) should be wrapped in  `<TabContext.Consumer>` like the `ThemeTogglerButton` in the docs? - So clicking on a `<Tab>` would update the state object on this `<Tabs>` component?

Reply:
No. `<TabNav>` will be told directly what tab is currently selected. Context is useful for components that don't have a direct relation to their parent tab.